src/core/task/Task.ts

```diff
@@ -667,10 +669,30 @@ export class Task extends EventEmitter<TaskEvents> implements TaskLike {
 				ts: Date.now(),
 			}
 
-			// If we have encrypted_content, embed it as the first content block on the assistant message.
+			// If we have reasoning text (from streaming), store it as a reasoning block
+			// This is the primary path for most providers (Anthropic, Gemini, Mistral, etc.)
+			if (reasoning) {
+				const reasoningBlock = {
+					type: "reasoning",
+					text: reasoning,
+					summary: reasoningSummary ?? ([] as any[]),
+				}
+
+				if (typeof messageWithTs.content === "string") {
+					messageWithTs.content = [
+						reasoningBlock,
+						{ type: "text", text: messageWithTs.content } satisfies Anthropic.Messages.TextBlockParam,
+					]
+				} else if (Array.isArray(messageWithTs.content)) {
+					messageWithTs.content = [reasoningBlock, ...messageWithTs.content]
+				} else if (!messageWithTs.content) {
+					messageWithTs.content = [reasoningBlock]
+				}
+			}
+			// If we have encrypted_content (OpenAI Native specific), embed it as the first content block
 			// This keeps reasoning + assistant atomic for context management while still allowing providers
 			// to receive a separate reasoning item when we build the request.
-			if (reasoningData?.encrypted_content) {
+			else if (reasoningData?.encrypted_content) {
```

Comment:
Super nitpick, but I'm not used to the else coming after a bunch of comments. Maybe we could put the comments inside of the block?